jscomp/syntax/ppx_entry.ml

```diff
@@ -270,9 +288,12 @@ let rec unsafe_mapper : Ast_mapper.mapper =
         | Pexp_extension (
             {txt = ("bs.re" | "re"); loc} , payload)
           ->
+          if validate_regexp payload then
           Exp.constraint_ ~loc
             (Ast_util.handle_raw loc payload)
-            (Ast_comb.to_js_re_type loc)            
+            (Ast_comb.to_js_re_type loc)
```

Comment:
also see ounit_cmd_tests.ml, we already have some infrastructure to test compilation error message